### PR TITLE
handle logs coming in from lambda functions

### PR DESCRIPTION
**Pull request by rgarcia**
https://clever.atlassian.net/browse/INFRA-2639

The code here previously assumed that data coming from a CWLogs subscription had the format from AWS Batch. This extends that logic to accomodate CWLogs data coming from Lambda.

I've added some tests, and also tested manually by:
- sending some data via the KPL to a kinesis stream
- sending some data via Lambda -> CWLogs into the same kinesis stream
- seeing kinesis-to-firehose (a user of this library) correctly parse the data coming through the stream, regardless of whether it came from the KPL or CWLogs

### CODE REVIEW

**Comment by rgarcia** [comment]
cc @nathanleiby

File: splitter/splitter.go [hunk 1]
```diff
@@ -67,42 +99,112 @@ const RFC3339Micro = "2006-01-02T15:04:05.999999-07:00"
...
+	ProgramName string
+	PID         int
+	Message     string
+}
+
+func (r RSysLogMessage) String() string {
+	// Adding an extra Microsecond forces `Format` to include all 6 digits within the micorsecond format.
+	// Otherwise, time.Format omits trailing zeroes. (https://github.com/golang/go/issues/12472)
+	return fmt.Sprintf(`%s %s %s[%d]: %s`,
+		r.Timestamp.Add(time.Microsecond).Format(RFC3339Micro),
```
Comment (nathanleiby) [inline splitter/splitter.go:127]: In the very unlikely case arrived with a timestamp of `x.999999` would adding a microsecond make this `<x+1>.000000`, resulting in the stringified timestamp losing its trailing 0s?

File: splitter/splitter_test.go [hunk 2]
```diff
@@ -125,3 +125,31 @@ func TestSplit(t *testing.T) {
...
+			},
+		},
+	}
+	lines := Split(input)
+	expected := [][]byte{
+		[]byte(`2017-06-26T23:32:23.285001+00:00 aws-lambda env--app/arn%3Aaws%3Aecs%3Aus-east-1%3A999988887777%3Atask%2F8edbd53f-64c7-4a3c-bf1e-efeff40f6512[1]: START RequestId: 8edbd53f-64c7-4a3c-bf1e-efeff40f6512 Version: 3`),
+		[]byte(`2017-06-26T23:32:23.285001+00:00 aws-lambda env--app/arn%3Aaws%3Aecs%3Aus-east-1%3A999988887777%3Atask%2F8edbd53f-64c7-4a3c-bf1e-efeff40f6512[1]: {"aws_request_id":"8edbd53f-64c7-4a3c-bf1e-efeff40f6512","level":"info","source":"app","title":"some-log-title"}`),
+	}
+	assert.Equal(t, expected, lines)
+}
```
Comment (nathanleiby) [inline splitter/splitter_test.go:155]: Is there a test yet that covers `splitDefault`?

**Review by nathanleiby — COMMENTED** [review]
looks great - thanks for cleaning this logic up

File: splitter/splitter_test.go (hunk 2 above)
Comment (rgarcia) [reply splitter/splitter_test.go:155]: c255bf8

File: splitter/splitter.go (hunk 1 above)
Comment (rgarcia) [reply splitter/splitter.go:127]: the timestamps from cwlogs are in milliseconds, so AFAICT it's not possible

Comment (nathanleiby) [reply splitter/splitter.go:127]: Excellent!

**Review by nathanleiby — APPROVED** [review]

**Review by xavi- — APPROVED** [review]
Nice.  Appreciate the general clean up as well.

**Pull request merged**